# Description of `Locations` table

In [1]:
import pandas as pd
import seaborn as sns

In [2]:
df = pd.read_csv('../tables_creation/final_tables/locations.csv')
df.head()

regname                areaname  \
0        Запорізька область   Мелітопольський район   
1       Хмельницька область     Красилівський район   
2  Дніпропетровська область  Петропавлівський район   
3       Чернівецька область              м.Чернівці   
4      Миколаївська область      Врадіївський район   

                      tername  KOATUU_2020         KATOTTG_2023 category  \
0                  с.Терпіння   2323085101  UA23080270010078454  village   
1                  м.Красилів   6822710100  UA68040210010032567     town   
2                 с.Дмитрівка   1223881501  UA12140170040016918  village   
3  Шевченківський район міста   7310100000  UA73060610010033137     town   
4                    с.Кумарі   4822383001  UA48080050190079797  village   

       region_name  
0        Zaporizka  
1      Khmelnytska  
2  Dnipropetrovska  
3     Chernivetska  
4      Mykolaivska

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29219 entries, 0 to 29218
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   regname       29219 non-null  object
 1   areaname      29219 non-null  object
 2   tername       29219 non-null  object
 3   KOATUU_2020   29219 non-null  int64 
 4   KATOTTG_2023  29219 non-null  object
 5   category      29219 non-null  object
 6   region_name   29219 non-null  object
dtypes: int64(1), object(6)
memory usage: 1.6+ MB


In [4]:
# print unique values for each column
for col in df.columns:
    print('Unique length in ', col+':\t', len(df[col].unique()))

Unique length in  regname:	 58
Unique length in  areaname:	 985
Unique length in  tername:	 18706
Unique length in  KOATUU_2020:	 18542
Unique length in  KATOTTG_2023:	 18542
Unique length in  category:	 6
Unique length in  region_name:	 59


Features:
- regname (Region name): 58 unique Region names: region, city of Kyiv, other countries names. This column can contain duplicates due to mistakes in the data, better to use only codes.
- areaname (Area name): 985 unique Area names: area, city, other countries city. This column can contain duplicates due to mistakes in the data, better to use only codes.
- tername (Territory name): 18706 unique Territory names: territory, city area, other countries city. This column can contain duplicates due to mistakes in the data, better to use only codes.
- KOATUU_2020 (KOATUU code): 18542 unique codes(a unique digital code for each administrative-territorial entity of Ukraine, used from January 1998 till January 2021): territory, city area, other countries city. KOATUU codes in Ukraine follow a six-digit format. The first two digits signify the region, the next two represent the district, and the following two denote the city or municipality.
- KATOTTG_2023 (KATOTTG code): 18542 unique KATOTTG codes(a unique digital code for each administrative-territorial entity of Ukraine, used from January 2021 after decentralization): territory, city area, other countries city.
- category (type of the settlement). Range of values:
    - city
    - town
    - settlement
    - urban village
    - village
    - abroad
- region_name (English name of the region): 59 unique Region names in English: region, city of Kyiv, other countries names. This column can contain duplicates due to mistakes in the data, better to use only codes.